fhir-persistence/src/main/java/org/linuxforhealth/fhir/persistence/FHIRPersistenceSupport.java

```diff
@@ -146,27 +150,42 @@ public static Instant getLastUpdatedFromResource(Resource resource) {
      * Get the lastUpdated time to use for the next version of a resource. If a current
      * version of the resource exists, pass its lastUpdated time as the currenLastUpdated
      * parameter (all times should be UTC).
+     * If the gap falls within a grace period of 2 seconds, then we make the new lastUpdated
+     * time equal the current time + 1 ms to maintain consistent ordering. Clusters should easily
+     * be able to maintain clocks synchronized within 2 seconds if they are configured with a
+     * reliable network time service.
      * @param currentLastUpdated
      * @return
      */
-    public static org.linuxforhealth.fhir.model.type.Instant getNewLastUpdatedInstant(Instant currentLastUpdated) {
+    public static org.linuxforhealth.fhir.model.type.Instant getNewLastUpdatedInstant(Instant currentLastUpdated) throws FHIRPersistenceException {
```

Comment:
i think we should add the @throws to the javadoc and mention that it throws if the currentLastUpdated is more than 2 seconds ahead

Reply:
done